web/src/pages/Dashboard.tsx

```diff
@@ -175,6 +175,13 @@ const StatsWrapper = () => {
     useRecoilState(selectedCity);
   const [totalSelectableCities, setTotalSelectableCities] =
     React.useState(selectedCities);
+  const { updateCityList } = useUpdateCityList();
+
+  React.useEffect(() => {
+    // This function will only be called once, when the component is first rendered
+    updateCityList();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I need this to avoid repetitive calls to fetch selected cities and fetch only on first render